checker.cpp

```diff
@@ -2,21 +2,92 @@
 #include <iostream>
 using namespace std;
 
+#define MIN_TEMP_THRESHOLD			0
+#define MAX_TEMP_THRESHOLD			45
+#define MIN_STATE_OF_CHARGE			20
+#define MAX_STATE_OF_CHARGE			80
+#define CHARGE_RATE					0.8
+
+
+class BatteryParams {
+private:
+	float temperature;
+	float soc;
+	float chargeRate;
+	float dischargeRate;
+	const float minSOC = MIN_STATE_OF_CHARGE;
+	const float maxSOC = MAX_STATE_OF_CHARGE;
+	const float minTemp = MIN_TEMP_THRESHOLD;
+	const float maxTemp = MAX_TEMP_THRESHOLD;
+	const float maxChargeRate = CHARGE_RATE;
+public:
+	BatteryParams(float temperature, float soc, float chargeRate);
+	void UpdateParams(float temperature, float soc, float chargeRate);
+	bool ValidateTemperature();
+	bool ValidateStateOfCharge();
+	bool ValidateChargeRate();
+};
+
+BatteryParams::BatteryParams(float inputTemp, float inputsoc, float inputchargeRate) {
+	temperature = inputTemp;
+	soc = inputsoc;
+	chargeRate = inputchargeRate;
+}
+
+void BatteryParams::UpdateParams(float inputTemp, float inputsoc, float inputchargeRate) {
+	temperature = inputTemp;
+	soc = inputsoc;
+	chargeRate = inputchargeRate;
+}
+
+bool BatteryParams::ValidateTemperature() {
+	if (temperature < minTemp || temperature > maxTemp) {
```

Comment:
This function is very similar to the next one. Introducing a warning level will lead to multiple updates. Reduce the duplication by abstracting the range check into a function and calling it with the parameter, & limits